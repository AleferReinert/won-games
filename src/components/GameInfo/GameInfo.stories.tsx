import type { StoryObj, Meta } from '@storybook/react'
import { within } from '@storybook/testing-library'
import { expect } from '@storybook/jest'
import GameInfoComponent from './GameInfo'
import mockGameInfo from './mock'

const meta: Meta<typeof GameInfoComponent> = {
  title: 'Components/GameInfo',
  component: GameInfoComponent,
  args: mockGameInfo
}

export default meta

type Story = StoryObj<typeof GameInfoComponent>

export const GameInfo: Story = {
  play: ({ canvasElement }) => {
    const canvas = within(canvasElement)
    const title = canvas.getByRole('heading', { name: /borderlands 3/i })
    const description = canvas.getByText(/now is hour to eliminate/i)
    const price = canvas.getByText('$215.00')
    const buttonAddToCart = canvas.getByRole('button', { name: /add to cart/i })
    const buttonWishlist = canvas.getByRole('button', { name: /wishlist/i })

    expect(title).toBeInTheDocument()
    expect(description).toBeInTheDocument()
    expect(price).toBeInTheDocument()
    expect(buttonAddToCart).toBeInTheDocument()
    expect(buttonWishlist).toBeInTheDocument()
  }
}
