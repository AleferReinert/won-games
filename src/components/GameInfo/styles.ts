import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'
import * as RibbonStyles from 'components/Ribbon/styles'
import * as ButtonStyles from 'components/Button/styles'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    padding: ${theme.spacings.medium} ${theme.spacings.small};
    position: relative;

    ${HeadingStyles.Wrapper} {
      font-size: ${theme.font.sizes.xlarge};
      margin-bottom: ${theme.spacings.xsmall};
    }

    ${RibbonStyles.Wrapper} {
      border-radius: ${theme.border.radius};
      font-size: ${theme.font.sizes.large};
      padding-left: ${theme.spacings.xsmall};
      padding-right: ${theme.spacings.xsmall};
      top: ${theme.spacings.medium};

      &::after {
        display: none;
      }
    }

    ${media.greaterThan('medium')`

        ${HeadingStyles.Wrapper} {
            font-size: ${theme.font.sizes.xxlarge};
        }

        ${RibbonStyles.Wrapper} {
            right: ${theme.spacings.xlarge};
            padding-left: ${theme.spacings.small};
            padding-right: ${theme.spacings.small};
        }
    `}
  `}
`
export const Description = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};

    ${media.greaterThan('medium')`
        max-width: 60%;
    `}
  `}
`

export const ButtonsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    row-gap: ${theme.spacings.xxsmall};

    ${media.greaterThan('medium')`
        flex-direction: row;
        justify-content: end;
        column-gap: ${theme.spacings.xxsmall};

        ${ButtonStyles.Wrapper} {
            width: auto;
        }
    `}
  `}
`
