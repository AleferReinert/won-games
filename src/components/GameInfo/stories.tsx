import { ComponentStory, ComponentMeta } from '@storybook/react'
import GameInfo from '.'
import mockGameInfo from './mock'
import theme from 'styles/theme'

export default {
  title: 'GameInfo',
  component: GameInfo,
  parameters: {
    layout: 'fullscreen'
  },
  args: mockGameInfo
} as ComponentMeta<typeof GameInfo>

export const Default: ComponentStory<typeof GameInfo> = (args) => (
  <div
    style={{
      margin: '0 auto',
      maxWidth: theme.grid.container,
      padding: '2.5rem'
    }}
  >
    <GameInfo {...args} />
  </div>
)
