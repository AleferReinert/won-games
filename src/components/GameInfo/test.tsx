import { screen } from '@testing-library/react'
import GameInfo from '.'
import { renderWithTheme } from 'utils/tests/helpers'
import props from './mock'

describe('<GameInfo />', () => {
  it('should render title, description, price and buttons', () => {
    const { container } = renderWithTheme(<GameInfo {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()
    expect(screen.getByText(props.description)).toBeInTheDocument()
    expect(screen.getByText('$' + props.price)).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /add to cart/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /wishlist/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
