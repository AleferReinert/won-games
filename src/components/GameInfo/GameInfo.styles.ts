import styled, { css } from 'styled-components'
import { cssMediaQuery } from 'utils/tests/helpers'
import * as ButtonStyles from 'components/Button/Button.styles'
import * as HeadingStyles from 'components/Heading/Heading.styles'
import * as RibbonStyles from 'components/Ribbon/Ribbon.styles'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    padding: ${theme.spacings.medium} ${theme.spacings.small};
    position: relative;

    ${HeadingStyles.Wrapper} {
      font-size: ${theme.font.sizes.xlarge};
      margin-bottom: ${theme.spacings.small};
    }

    ${RibbonStyles.Wrapper} {
      border-radius: ${theme.border.radius};
      font-size: ${theme.font.sizes.large};
      padding-left: ${theme.spacings.xsmall};
      padding-right: ${theme.spacings.xsmall};
      top: ${theme.spacings.medium};

      &::after {
        display: none;
      }
    }

    ${cssMediaQuery.greaterThan(theme.breakpoint.small)} {
      ${HeadingStyles.Wrapper} {
        font-size: ${theme.font.sizes.xxlarge};
      }

      ${RibbonStyles.Wrapper} {
        right: ${theme.spacings.xlarge};
        padding-left: ${theme.spacings.small};
        padding-right: ${theme.spacings.small};
      }
    }
  `}
`
export const Description = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};

    ${cssMediaQuery.greaterThan(theme.breakpoint.small)} {
      font-size: ${theme.font.sizes.large};
      max-width: 80rem;
    }
  `}
`

export const ButtonsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    row-gap: ${theme.spacings.xxsmall};

    ${cssMediaQuery.greaterThan(theme.breakpoint.small)} {
      flex-direction: row;
      justify-content: end;
      column-gap: ${theme.spacings.xxsmall};

      ${ButtonStyles.Wrapper} {
        min-width: auto;
      }
    }
  `}
`
