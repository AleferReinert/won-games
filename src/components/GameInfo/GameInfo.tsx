import {
  AddShoppingCart,
  FavoriteBorder
} from '@styled-icons/material-outlined'
import Button from 'components/Button/Button'
import Heading from 'components/Heading/Heading'
import Price from 'components/Price/Price'
import * as S from './GameInfo.styles'

export type GameInfoProps = {
  title: string
  description: string
  price: string
}

const GameInfo = ({ title, description, price }: GameInfoProps) => {
  return (
    <S.Wrapper data-testid='gameInfoComponent'>
      <Heading color='black'>{title}</Heading>

      <S.Description>{description}</S.Description>

      <Price price={price} />

      <S.ButtonsWrapper>
        <Button icon={<AddShoppingCart />} full>
          Add to cart
        </Button>
        <Button icon={<FavoriteBorder />} full variant='link'>
          Wishlist
        </Button>
      </S.ButtonsWrapper>
    </S.Wrapper>
  )
}

export default GameInfo
