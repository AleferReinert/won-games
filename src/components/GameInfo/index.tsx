import Heading from 'components/Heading'
import * as S from './styles'
import Ribbon from 'components/Ribbon'
import Button from 'components/Button'
import {
  AddShoppingCart,
  FavoriteBorder
} from '@styled-icons/material-outlined'

export type GameInfoProps = {
  title: string
  description: string
  price: string
}

const GameInfo = ({ title, description, price }: GameInfoProps) => {
  return (
    <S.Wrapper>
      <Heading color='black' lineBottom>
        {title}
      </Heading>

      <S.Description>{description}</S.Description>

      <Ribbon color='secondary'>{'$' + price}</Ribbon>

      <S.ButtonsWrapper>
        <Button icon={<AddShoppingCart />} fullWidth>
          Add to cart
        </Button>
        <Button icon={<FavoriteBorder />} fullWidth minimal>
          Wishlist
        </Button>
      </S.ButtonsWrapper>
    </S.Wrapper>
  )
}

export default GameInfo
