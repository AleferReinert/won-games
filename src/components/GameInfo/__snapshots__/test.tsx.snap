// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfo /> should render title, description, price and buttons 1`] = `
.c2 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c2::after {
  content: '';
  position: absolute;
  left: 0;
  bottom: -0.5rem;
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c5 {
  display: inline-block;
  color: #FAFAFA;
  font-weight: 600;
  position: absolute;
  height: 3.3rem;
  font-size: 1.4rem;
  line-height: 3.3rem;
  padding: 0 4rem;
  top: 5rem;
  right: -1.4rem;
  background-color: #3CD3C1;
}

.c5::after {
  content: '';
  height: 0;
  display: block;
  position: absolute;
  right: 0;
  border-style: solid;
}

.c5::after {
  bottom: -0.7rem;
  border-width: 0.4rem 0.7rem;
}

.c5::after {
  border-color: #3CD3C1 transparent transparent #3CD3C1;
  -webkit-filter: brightness(75%);
  filter: brightness(75%);
}

.c8 {
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  cursor: pointer;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0 3.2rem;
  width: 100%;
}

.c8:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c8 svg {
  width: 1.5rem;
}

.c8 svg + span {
  margin-left: 0.8rem;
}

.c10 {
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  cursor: pointer;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0 3.2rem;
  width: 100%;
  background: transparent;
  color: #F231A5;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
}

.c10:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c10 svg {
  width: 1.5rem;
}

.c10 svg + span {
  margin-left: 0.8rem;
}

.c10:hover {
  background: transparent;
  color: #b20b6f;
}

.c0 {
  background-color: #FAFAFA;
  padding: 3.2rem 2.4rem;
  position: relative;
}

.c0 .c1 {
  font-size: 2.0rem;
  margin-bottom: 1.6rem;
}

.c0 .c4 {
  border-radius: 0.4rem;
  font-size: 1.8rem;
  padding-left: 1.6rem;
  padding-right: 1.6rem;
  top: 3.2rem;
}

.c0 .c4::after {
  display: none;
}

.c3 {
  color: #8F8F8F;
  font-size: 1.4rem;
  margin-bottom: 2.4rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 0.8rem;
}

.c9 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c0 .c1 {
    font-size: 2.8rem;
  }

  .c0 .c4 {
    right: 4.8rem;
    padding-left: 2.4rem;
    padding-right: 2.4rem;
  }
}

@media (min-width:768px) {
  .c3 {
    max-width: 60%;
  }
}

@media (min-width:768px) {
  .c6 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: end;
    -webkit-justify-content: end;
    -ms-flex-pack: end;
    justify-content: end;
    -webkit-column-gap: 0.8rem;
    column-gap: 0.8rem;
  }

  .c6 .c7 {
    width: auto;
  }
}

<div
  class="c0"
>
  <h2
    class="c1 c2"
    color="black"
  >
    Borderlands 3
  </h2>
  <p
    class="c3"
  >
    Now is hour to eliminate Handsome Jack. Gather your friends to run in the world collecting millions of items.
  </p>
  <div
    class="c4 c5"
    color="secondary"
  >
    $215.00
  </div>
  <div
    class="c6"
  >
    <button
      class="c7 c8"
    >
      <svg
        aria-hidden="true"
        class="c9"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4l-3.87 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2z"
        />
      </svg>
      <span>
        Add to cart
      </span>
    </button>
    <button
      class="c7 c10"
    >
      <svg
        aria-hidden="true"
        class="c9"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
      <span>
        Wishlist
      </span>
    </button>
  </div>
</div>
`;
